website/docs/concepts/components/children.md

```diff
@@ -124,3 +126,52 @@ impl Component for List {
 }
 ```
 
+### Optional typed child
+You can also have a single optional child component of a specific type too: 
+
+```rust
+use yew::prelude::*;
+use yew::virtual_dom::VChild;
+
+
+#[derive(Clone, Properties)]
+pub struct PageProps {
+    #[prop_or_default]
+    pub sidebar: Option<VChild<PageSideBar>>,
+}
+
+struct Page {
+    props: PageProps,
+}
+
+impl Component for Page {
+    type Properties = PageProps;
+    // ...
+
+    fn view(&self) -> Html {
+        html! {
+            <div class="page">
+                { self.props.sidebar.clone().map(Html::from).unwrap_or_default() }
+                // ... page content
+            </div>
+        }
+    }
+}
+```
+
+The page component can be called either with the sidebar or without: 
+
+```rust
+    // Page without sidebar
+    html! {
+        <Page />
+    }
+
+    // Page with sidebar
+    html! {
+        <Page sidebar={html_nested!{
+            <PageSideBar />
+        }}>
+        </Page>
```

Comment:
If you want to do it inline, I'd prefer to keep the style consistent:
```suggestion
        <Page sidebar=html_nested! { <PageSideBar /> } />
```

or if you prefer:

```suggestion
        <Page sidebar=html_nested! {
            <PageSideBar />
        } />
```